src/DocumentDB.ChangeFeedProcessor/LeaseManagement/DocumentServiceLeaseStoreManager.cs

```diff
@@ -34,6 +34,8 @@ internal class DocumentServiceLeaseStoreManager : ILeaseStoreManager
         private readonly IDocumentServiceLeaseUpdater leaseUpdater;
         private readonly ILeaseStore leaseStore;
 
+        private bool isPartitionedByLeasePk = false;
```

Comment:
This is not needed at this layer, please remove. This should be responsibility of the request options factory.